agent/helpers/cache_handler.c

```diff
@@ -785,13 +788,21 @@ release_cached_resources(unsigned int regNo, void *clientargs)
              *   least one active cache.
              */
             if (netsnmp_cache_check_expired(cache)) {
-                if(! (cache->flags & NETSNMP_CACHE_DONT_FREE_EXPIRED))
+                if(! (cache->flags & NETSNMP_CACHE_DONT_FREE_EXPIRED)) {
                     _cache_free(cache);
+                    if (cache->free_cache && !cache->timer_id)
+                        do_trim = 1;
+                }
             } else {
                 cache_outstanding_valid = 1;
             }
         }
     }
+
+    if (do_trim)
+        /* Release freed memory back to the system */
+        malloc_trim(0);
+
```

Comment:
All systems I know of that run Net-SNMP support virtual memory. Is calling malloc_trim() on such systems useful?

Reply:
Yes, the patch was actually tested on Linux.